imyelo/docsify-pagination
# Style updates for themes (fixes #3)

Styles:

- Replaced fixed `color` values with `currentColor`
- Replaced fixed `font-size` values with relative units
- Replaced fixed `margin` values with relative units
- Replaced top margin `border` color with rgba value
- Replaced background svg icons with inline svg elements to allow styling via CSS
- Replaced svg `<path>` with `<polyline>` to allow styling using stroke properties
- Moved "themeable" svg styles to CSS (`stroke`, `stroke-width`, etc.)
- Fixed opacity on "previous" pagination item when a "Next" link is not present (should be full opacity when a "Next" link is not present)
- Fixed `:hover` effect on "previous" link (both `pagination-item-label` and `pagination-item-title` now respond to hover state)

Markup:

- Changed icon position (icons now "outside" rather than "inside" of previous/next text).
- Updated logic in `pagination.js` to only render `div.pagination-item` elements if a link will be displayed. This allows styling items based on the number of `div.pagination-item` elements present.
- Added additional theme (`dark.css`) and `<style>` for testing theme compatibility. These are commented out by default. Uncomment to test.

JavaScript:

- Fixed error when $docsify is not defined

Misc:

- Removed `postcss-inline-svg` dependency
- Removed `postcss-svgo` dependency
- Removed `uglify-js` dependency (no need to uglify/minify since CSS is inject on the client)
- Removed `postcss.config.js` (not needed since no plugins are used)

Testing:

1. Enable `dark.css` theme in `index.html`. Notice how all pagination elements adopt styles from the theme.
1. Enable individual rules in the `<style>` block. Notice how theme authors can now easily modify styles for their theme, including SVG properties.

Happy to answer any questions you may have, and thanks for reviewing the PR!
